### PR TITLE
Improvements to plugin config validation, plugin imports, and serving plugin config from Deck.

### LINKED ISSUE #13951: Deck plugin config validation fails
In order to reuse /test logic for presubmit reruns from the frontend (see #13321), deck needs access to the plugin config. Currently, it throws a fatal error on validation because we don't import all the plugins. We either need to avoid doing this specific validation in deck, import the plugins, or find some other solution.

### PR DESCRIPTION
**Pull request by cjwagner**
I made a different few semi-related fixes and improvements in this PR, but I separated them by commit so I recommend reviewing commit-by-commit.

fixes #13951  (and will unblock #14401)
I tested Deck with `runlocal` and confirmed that it can now successfully load the plugin config.

/kind bug
/assign @stevekuznetsov 
/cc @alvaroaleman @fejta

### CODE REVIEW

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -1327,9 +1336,10 @@ func canTriggerJob(user string, pj prowapi.ProwJob, cfg *prowapi.RerunAuthConfig
 // POST request, and the user has the necessary permissions. Otherwise, it writes the config
 // for a new job but does not trigger it.
 func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool, cfg authCfgGetter, goa *githuboauth.Agent, ghc githuboauth.GitHubClientGetter, cli prowgithub.RerunClient, pluginAgent *plugins.ConfigAgent) http.HandlerFunc {
+	log := logrus.WithField("handler", "/rerun")
```
Comment (alvaroaleman) [inline prow/cmd/deck/main.go]: Nit: Would it make sense to define the loggers at the router level and pass them in? That way:
* If someone changes the route for something, they are likely to also change the log accordingly
* If someone adds a new handler, they will hopefully also pass in a logger with a `handler` field

File: prow/plugins/plugins.go [hunk 2]
```diff
@@ -19,11 +19,12 @@ package plugins
 import (
 	"errors"
 	"fmt"
-	"github.com/prometheus/client_golang/prometheus"
 	"io/ioutil"
 	"sync"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (fejta) [inline prow/plugins/plugins.go]: Should be part of the next block

File: prow/cmd/deck/main.go (hunk 1 above)
Comment (cjwagner) [reply prow/cmd/deck/main.go]: Done.

File: prow/plugins/plugins.go (hunk 2 above)
Comment (cjwagner) [reply prow/plugins/plugins.go]: Done.

**Comment by cjwagner** [comment]
Comments addressed.

**Comment by cjwagner** [comment]
Conquered bazel. PTAL
/hold cancel

**Comment by Katharine** [comment]
As of this PR (specifically, as of 2d918b69dafa9d1cb92c4f7be4bad27687473c77), I cannot run deck locally:

```{"component":"deck","error":"job pull-cri-containerd-build failed to merge presets for podspec: env var duplicated in pod spec: GOOGLE_APPLICATION_CREDENTIALS","file":"prow/cmd/deck/main.go:293","func":"main.main","level":"fatal","msg":"Error starting config agent.","time":"2019-10-01T09:45:56-07:00"}```

(the exact error varies, but it's always about merging presets. Sometimes it's `GOPROXY` instead of `GOOGLE_APPLICATION_CREDENTIALS`)

**Pull request merged**